### PR TITLE
#10: Dialect name list now assembled at runtime.

### LINKED ISSUE #10: Automate adding a dialect name to the factory list
## Problem
When we create a new dialect we have to add a dialect's name manually to the list in the method `getSupportedAdapterNames()` in the JdbcAdapterFactory class.

## Solution
Automate this process. 

This ticket was moved from: https://github.com/exasol/virtual-schema-common-java/issues/115

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #10.

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dialects/SqlDialectRegistry.java [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
 
 import java.sql.Connection;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.Set;
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dialects/SqlDialectRegistry.java]: Do you do these imports on purpose? it's more than 3, so it's usually a star

File: src/main/java/com/exasol/adapter/dialects/SqlDialectRegistry.java [hunk 2]
```diff
@@ -40,21 +46,33 @@ public void loadSqlDialectFactories() {
...
      *
-     * @return list of adapter factories
+     * @return list of dialect factories
      */
     public List<SqlDialectFactory> getRegisteredAdapterFactories() {
         return new ArrayList<>(this.registeredFactories.values());
     }
 
+    /**
+     * Get a list of the name of all registered SQL dialect adapters
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dialects/SqlDialectRegistry.java]:
```suggestion
     * Get a list of the name of all registered SQL dialect adapters.
```

File: src/main/java/com/exasol/adapter/dialects/SqlDialectRegistry.java (hunk 1 above)
Comment (redcatbear) [reply src/main/java/com/exasol/adapter/dialects/SqlDialectRegistry.java]: Fixed formatter settings.

File: src/main/java/com/exasol/adapter/dialects/SqlDialect.java [hunk 3]
```diff
@@ -52,14 +52,16 @@
      * Dialects that support a single database should not use <code>AUTO_DETECT</code> because this unnecessarily costs
      * performance.
      */
-    public enum StructureElementSupport {
+    public enum StructureElementSupport
+    {
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dialects/SqlDialect.java]: Is it the right formatter setting?

Comment (redcatbear) [reply src/main/java/com/exasol/adapter/dialects/SqlDialect.java]: No.

Comment (redcatbear) [reply src/main/java/com/exasol/adapter/dialects/SqlDialect.java]: Fixed.

**Pull request merged**